### DISCUSSION TITLE
Streamlit Perspective Component

### DISCUSSION THREAD

**Discussion by ccarli** [perspective-dev/perspective#2529] [Ideas]
Hello

I am using frequently the [streamlit](https://streamlit.io/) framework in order to publish and share apps, and have been trying to include in these applications some perspective tables and charts.

Is it planned in the project roadmap to support an "official" streamlit extension (similarly to this [aggrid component](https://github.com/PablocFonseca/streamlit-aggrid) ). It would be very fitting bridge between these two projects to allow neat data exploration !

Thanks

**Comment by timkpaine** [comment]
We have piloted supporting components in these kinds of frameworks in the past (e.g. a [component for plotly dash](https://github.com/perspective-community/perspective-dash-component)) and it exceeds our capacity as maintainers, we are quite a small team. So unfortunately I think for the foreseeable future there will be no plan for an official component.

That being said, we're happy to help the community support these components, and some folks have built components for other frameworks, e.g. [Panel](https://panel.holoviz.org/reference/panes/Perspective.html). Given the relatively simple interface Perspective exposes on both the Python and JS side, which doesn't require any particular JS framework or Python web server framework, it shouldn't be so hard for *someone* to build such a component for Streamlit. Its just not something we (the maintainers of perspective) have the capacity to do at this time.
